examples/src/main/java/com/mapbox/navigation/examples/core/InstructionViewActivity.kt

```diff
@@ -0,0 +1,429 @@
+    }
+
+    override fun onMapReady(mapboxMap: MapboxMap) {
+        Timber.d("onMapReady")
+        this.mapboxMap = mapboxMap
+        mapboxMap.moveCamera(CameraUpdateFactory.zoomTo(15.0))
+
+        mapboxMap.addOnMapLongClickListener { latLng ->
+            Timber.d("onMapLongClickListener position=%s", latLng)
+            destination = latLng
+            locationComponent?.lastKnownLocation?.let { originLocation ->
+                mapboxNavigation.requestRoutes(
+                    RouteOptions.builder().applyDefaultParams()
+                        .accessToken(Utils.getMapboxAccessToken(applicationContext))
+                        .coordinates(originLocation.toPoint(), null, latLng.toPoint())
+                        .alternatives(true)
+                        .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
+                        .build(),
+                    routesReqCallback
+                )
+            }
+            true
+        }
+
+        mapboxMap.setStyle(Style.MAPBOX_STREETS) { style ->
+            locationComponent = mapboxMap.locationComponent.apply {
+                activateLocationComponent(
+                    LocationComponentActivationOptions.builder(this@InstructionViewActivity, style)
+                        .build()
+                )
+                cameraMode = CameraMode.TRACKING
+                isLocationComponentEnabled = true
+            }
+
+            navigationMapboxMap = NavigationMapboxMap(mapView, mapboxMap).also {
+                it.addProgressChangeListener(mapboxNavigation)
+                it.setCamera(DynamicCamera(mapboxMap))
+            }
+        }
+    }
+
+    // InstructionView Feedback Bottom Sheet listener
+    override fun onFeedbackSelected(feedbackItem: FeedbackItem?) {
+        feedbackItem?.let { feedback ->
+            mapboxMap?.snapshot { snapshot ->
+                MapboxNavigation.postUserFeedback(
+                    TelemetryUtils.obtainUniversalUniqueIdentifier(),
+                    feedback.feedbackType, feedback.description, encodeSnapshot(snapshot)
+                )
+            }
+        }
+    }
+
+    override fun onFeedbackDismissed() {
+        // do nothing
+    }
+
+    private fun initNavigation() {
+        val accessToken = Utils.getMapboxAccessToken(this)
+        mapboxNavigation = if (Utils.shouldSimulateRoute(applicationContext)) {
+            Timber.d("initNavigation simulate route")
+            MapboxNavigation(
+                applicationContext,
+                accessToken,
+                MapboxNavigation.defaultNavigationOptions(this, accessToken),
+                replayRouteLocationEngine
+            )
+        } else {
+            Timber.d("initNavigation location engine")
+            MapboxNavigation(
+                applicationContext,
+                accessToken,
+                MapboxNavigation.defaultNavigationOptions(this, accessToken)
+            )
+        }
+    }
```

Comment:
Let's just use `ReplayRouteLocationEngine` by default.

Reply:
`ReplayRouteLocationEngine` would simulate routeProgress even if you don't check `SimulateRoute`.  Remember the idea of this activity to show how to use `InstructionView`, not to show how to handle toggle simulate route or anything related to simulation of routeProgress

Reply:
> `ReplayRouteLocationEngine` would simulate routeProgress even if you don't check `SimulateRoute`. 

I understand this. I mean what's the side effect of having this logic?

Reply:
It's good to have this when we use `mockLocation` app. But, in this case since a different activity already has handled this logic and shows its use as well as replayRouteLocationEngine doesn't needs it, I don't see any point of why we should have this logic.